pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImpl.java

```diff
@@ -340,6 +344,53 @@ public Consumer subscribe(String topic, String subscription, ConsumerConfigurati
         return consumerSubscribedFuture;
     }
 
+    public CompletableFuture<Consumer> subscribeAsync(Pattern topicsPattern,
```

Comment:
Take a look at the refactored code that is in master already. There is a single `subscribeAsync(ConsumerConfigurationData conf)` method that is used for all the variations of the subscribe. 

Then, for this regex subscribe, we should reuse the same code that subscribe to the list of topics, just adding the extra pre-step of figuring out the list.

Reply:
Thanks, besides subscribe the list of topics, we need to pass in `pattern` in to subscribe, so the consumer will keep the pattern for later usage.